### ISSUE TITLE
Spike: Twitter advanced search documentation updates

### ISSUE THREAD

**Issue by lacyg4** [masa-finance/masa-oracle#399]
## Problem 
Users of the Twitter endpoint want more granular search capabilities. Our API supports advanced search this but its not documented. Advanced search queries can be constructed here: https://x.com/search-advanced?lang=en

We currently do not document examples of advanced search in the swagger documentation or developer documentation so people understand the full functionality of the twitter search API. 

Here are some examples of advanced search queries: 

```
curl -X 'POST' \
  'http://localhost:8080/api/v1/data/twitter/tweets/recent' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "query": "Masa Moon (from:getmasafi) until:2022-01-01 since:2006-01-01",
  "count": 10
}'
```

```
curl -X 'POST' \
  'http://localhost:8080/api/v1/data/twitter/tweets/recent' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "query": "(Masa OR CoinList)(from:getmasafi) until:2022-01-01 since:2006-01-01",
  "count": 10
}'
```

These developer documentation can be found here: https://help.x.com/en/using-x/x-advanced-search

## Customer feedback
https://masafinance.slack.com/archives/C078JCDS81E/p1719870998918199
https://masafinance.slack.com/archives/C073K1YP8CR/p1717494858189539

## Acceptance criteria:

1. Documentation Structure:
   - [ ] Create a single, dedicated "Advanced Search" section in the API documentation
   - [ ] Implement a dropdown or tabbed interface for users to select different search types TBD

2. Search Type Documentation:
   - [ ] For each search type, provide:
     - [ ] A brief description of the search type
     - [ ] The correct syntax or format to use
     - [ ] At least one clear example of usage
     - [ ] Any relevant limitations or special considerations

6. Testing:
   - [ ] Conduct user testing to ensure the documentation is easy to understand and use
   - [ ] Verify all examples are correct and functional

### Advance search to support:

- [ ] Exact phrase search
- [ ] Hashtag search
- [ ] Mention search (@ symbol)
- [ ] From user search
- [ ] To user search
- [ ] Language-specific search
- [ ] Date range search
- [ ] Retweets filtering
- [x] Minimum likes filter
- [x] Minimum retweets filter
- [x] Keyword exclusion (NOT operator)
- [x] OR operator for multiple terms
- [x] Geo-location based search
- [x] URL inclusion search
- [x] Question tweets filter
- [x] Safe search mode

## Possible solutions 
1. Document each case in swagger as an example (expensive)
2. Document each search query and have a couple of examples in dev docs

**Comment by teslashibe** [comment]
We may want to think about renaming the `/recent` endpoint

![Image](https://github.com/masa-finance/masa-oracle/assets/34052452/4bb65cdc-b437-4a3b-8d23-19c4a1fc0a58)

**Comment by giovaroma** [comment]
@teslashibe updated the advance search option from Twitter and the acceptance criteria, please review the list of advance search options and let me know if this looks correct or we should remove some.

**Comment by teslashibe** [comment]
@giovaroma this looks fantastic - let move forward with this. 

I have added this to a release in the roadmap here: https://github.com/masa-finance/roadmap/issues/2 (cc @mudler @H34D @Luka-Loncar)

I added this to no-status in the protocol board and it can be groomed on Monday

**Comment by giovaroma** [comment]
Wonderful, moving tickets down the pipeline @teslashibe

**Comment by mudler** [comment]
potentially a duplicate of https://github.com/masa-finance/masa-oracle/issues/362 ?

**Comment by mudler** [comment]
Exact match documentation depends on #351

**Comment by Luka-Loncar** [comment]
Exact match and advanced search should be tested first. @nolanjacobson

**Comment by mudler** [comment]
This has been done and QA'd, follow-up https://github.com/masa-finance/masa-oracle/issues/459
